jvmhs/src/Jvmhs/Analysis/Reduce.hs

```diff
@@ -11,4 +12,94 @@ This module defines the a class hierarchy analysis.
 module Jvmhs.Analysis.Reduce
   where
 
-import Control.Lens
+import            Control.Monad.IO.Class
+import            Control.Lens
+import            Jvmhs
+import            Jvmhs.Data.Code
+import            Data.Foldable
+import            Data.Traversable
+import            Data.Int
+import            Data.Word
+import            Data.Text.Internal
+import            Data.List
+
+import qualified Data.Set                             as S
+
+import qualified Language.JVM                         as B
+import qualified Language.JVM.Attribute.Code          as B
+import qualified Language.JVM.Attribute.StackMapTable as B
+import qualified Language.JVM.Constant                as B
+import qualified Language.JVM.Stage                   as B
+
+
+findMethod :: Class -> MethodId -> Maybe Method
+findMethod cls mid =
+  preview (classMethods.folded.filtered(view $ toMethodId.to(== mid))) cls
+
+
+javaLangObject::B.ClassName
+javaLangObject = B.ClassName {B.classNameAsText = "java/lang/Object"}
+
+reduceInterface :: [FilePath] -> FilePath -> ClassName -> IO ()
```

Comment:
Split this function in two parts.

- One that finds the unused interfaces
```haskell
findUnusedInterfaces :: 
    (Foldable t, MonadClassPool m) 
  => t ClassName
  -- ^ A world to search for interfaces
  -> m (M.Map ClassName [ClassName])
  -- ^ A map from interfaces to the parrents of those interfaces.
```
- One that removes unused interfaces from a single class.
```haskell
inlineInterfaces :: 
  M.Map ClassName [ClassName]
  -- ^ Collection of interfaces to inline
  -> Class 
  -> Class
  -- ^ If class has interface, replace it with the parrents of that interface
```
- And then a combination of those two.
```haskell
reduceInterfaces :: 
    (Foldable t, MonadClassPool m) 
  => t ClassName
  -> m ()
reduceInterfaces world = do 
   interfaces <- findUnusedInterfaces world
   forM_ world $ \cn -> modifyClass cn (inlineInterfaces interfaces)
```